packages/headless/src/features/product-recommendations/product-recommendations-helpers.ts

```diff
@@ -0,0 +1,116 @@
+import {isNullOrUndefined} from '@coveo/bueno';
+import {ProductRecommendation} from '../../product-listing.index';
+import {isArray} from '../../utils/utils';
+import {ProductRecommendationTemplateCondition} from './product-recommendations-templates';
+
+/**
+ * Extracts a property from a product recommendation object.
+ * @param productRecommendation (ProductRecommendation) The target product recommendation.
+ * @param property (string) The property to extract.
+ * @returns (unknown) The value of the specified property in the specified product recommendation, or null if the property does not exist.
+ */
+export const getProductRecommendationProperty = (
+  productRecommendation: ProductRecommendation,
+  property: string
+) => {
+  const anyProductRecommendation = productRecommendation as unknown as Record<
+    string,
+    unknown
+  >;
+  if (!isNullOrUndefined(anyProductRecommendation[property])) {
+    return anyProductRecommendation[property];
+  }
```

Comment:
Are these null or undefined checks necessary if every template goes through the schema in the manager build function ([line](https://github.com/coveo/ui-kit/blob/14ae1b810cff3da153fd5d9f53f8a9ad92d4a402/packages/headless/src/features/product-recommendations/product-recommendations-templates-manager.ts#L62)) Will templates be used used outside of the manager?